src/test/java/gameframework/motion/overlapping/OverlapProcessorDefaultImplTest.java

```diff
@@ -36,8 +38,52 @@ public void setGameData(GameData data) {
 			}
 		});
 	}
+	
+	@Test
+	public void intersectionBetweenOverlapSObjectTest() {
+		int width1 = 10;
+		int height1 = 20;
+		Overlappable overlappable1 = createOverlappableMovable(0, 0, width1, height1);
+		Overlappable overlappable2 = createOverlappable(1,1,width1, height1);
+		Shape targetShape = overlappable2.getBoundingBox();
+		Vector<Overlap> overlaps = new Vector<Overlap>();
+		this.overlapProcessor.overlapRectangle(overlappable1, overlaps, overlappable2, targetShape);
+		assertSame(1, overlaps.size());
```

Comment:
the fact that 2 numbers with the same value are the same is an implementation detail of Java. Please use `assertEquals` instead